activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -427,6 +427,13 @@ def new_column_definition(name, type, **options) # :nodoc:
 
       private
         def create_column_definition(name, type, options)
+          options.assert_valid_keys(
+            :after, :array, :as, :auto_increment, :charset, :collation, :comment,
+            :default, :first, :force, :foreign_key, :id, :index, :limit, :null,
+            :options, :polymorphic, :precision, :primary_key, :scale, :stored,
+            :temporary, :type, :unique, :unsigned
```

Comment:
A thought instead of hardcoding the list here: if these were defined on the adapter class (this here being the abstract adapter), then each database could have its own set of allowed options. (`def SubclassAdapter.valid_keys; super + [:custom_option]`)